src/main/java/frc/robot/subsystems/drive/TalonSwerveModule.java

```diff
@@ -0,0 +1,171 @@
+    toApply.CurrentLimits.SupplyCurrentLimitEnable = true;
+    toApply.CurrentLimits.StatorCurrentLimit =
+        TalonDriveVars.DRIVING_MOTOR_STATOR_TELEOP_CURRENT_LIMIT_AMPS;
+    toApply.CurrentLimits.StatorCurrentLimitEnable = true;
+    toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+    toApply.Slot0.kP = TalonDriveVars.DRIVING_P;
+    toApply.Slot0.kI = TalonDriveVars.DRIVING_I;
+    toApply.Slot0.kD = TalonDriveVars.DRIVING_D;
+    toApply.Slot0.kV = TalonDriveVars.DRIVING_FF;
+    drivingTalon.getConfigurator().apply(toApply);
+    final double fastUpdateFrequencyHz = 50.0; 
+    final double slowUpdateFrequencyHz = 50.0;
+    drivingTalon.getPosition().setUpdateFrequency(fastUpdateFrequencyHz);
+    drivingTalon.getVelocity().setUpdateFrequency(fastUpdateFrequencyHz);
+    drivingTalon.getClosedLoopProportionalOutput().setUpdateFrequency(slowUpdateFrequencyHz);
+    drivingTalon.getClosedLoopDerivativeOutput().setUpdateFrequency(slowUpdateFrequencyHz);
+    drivingTalon.getClosedLoopIntegratedOutput().setUpdateFrequency(slowUpdateFrequencyHz);
+    drivingTalon.getClosedLoopFeedForward().setUpdateFrequency(slowUpdateFrequencyHz);
+    drivingTalon.optimizeBusUtilization();
+    drivingTalon.setPosition(0);
+
+    //turning talon
+    toApply.Feedback.SensorToMechanismRatio =
+        TalonDriveVars.SENSOR_TO_MECHANISM_RATIO_TURNING;
+    toApply.CurrentLimits.SupplyCurrentLimit =
+        TalonDriveVars.TURNING_MOTOR_SUPPLY_CURRENT_LIMIT_AMPS;
+    toApply.CurrentLimits.StatorCurrentLimit =
+        TalonDriveVars.TURNING_MOTOR_STATOR_TELEOP_CURRENT_LIMIT_AMPS;
+    toApply.Slot0.kP = TalonDriveVars.TURNING_P;
+    toApply.Slot0.kI = TalonDriveVars.TURNING_I;
+    toApply.Slot0.kD = TalonDriveVars.TURNING_D;
+    toApply.Slot0.kV = TalonDriveVars.TURNING_FF;
+    toApply.Feedback.FeedbackRemoteSensorID = turningTalonEncoder.getDeviceID();
+    toApply.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.RemoteCANcoder;
+    turningTalon.getConfigurator().apply(toApply);
+  }
+
+  /** Ensures the value a is in [0, b) */
+  public static double mod(double a, double b) {
+    double r = a % b;
+    if (r < 0) {
+      r += b;
+    }
+    return r;
+  }
+
+  /** Consider zeroing based on absolute values */
+  public void considerZeroingEncoder(){
+    if(Math.abs(turningTalonEncoder.getAbsolutePosition().getValueAsDouble()) < 0.01){
+      return;
+    }
+    if(Math.abs(turningTalonEncoder.getAbsolutePosition().getValueAsDouble()-mod(turningTalon.getPosition().getValueAsDouble(), 1)) < TalonDriveVars.TURNING_ENCODER_ZERO_THRESHOLD_ROTATION){
+      turningTalon.setPosition(turningTalonEncoder.getAbsolutePosition().getValueAsDouble());
+    }
+  }
+
+  /** Get the current state of the module */
+  public SwerveModuleState getState(){
+    return new SwerveModuleState(
+      drivingTalon.getVelocity().getValueAsDouble(),
+      new Rotation2d(turningTalonEncoder.getAbsolutePosition().getValueAsDouble())
```

Comment:
this should be Rotation2d.fromRotations() (i think?) bc i think default is radians

Reply:
also shouldn't this be relative encoder bc last year we said:
new Rotation2d(turningRelativeEncoder.getPosition())

Reply:
done